### DISCUSSION TITLE
Adding back interesting 1.4 "image" functions

### DISCUSSION THREAD

**Discussion by sfederici** [ScratchAddons/ScratchAddons#3568] [Ideas]
I would love to be able to use in Scratch 3 some very useful Scratch 1.4 "image" functions (maybe useful just for Scratch teachers? I'm not sure, but, as a teacher, when I have taught Scratch to young kids and to teachers they were really useful) you had in your right-click menu when clicking the Stage or sprites on the Stage, namely:
- when right-clicking the Stage: 1) "grab screen region for new sprite" and 2) "save picture of Stage"
- when right-clicking a sprite on the Stage: 1) "grab screen region for new costume"

**Comment by cobaltt7** [comment]
In edge at least, exporting the stage is already possible, as all canvas elements by default have "Export image" right click options.

What is grabbing a screen region for new sprite?


________________________________
From: sfederici ***@***.***>
Sent: Sunday, September 26, 2021 1:43:11 PM
To: ScratchAddons/ScratchAddons ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [ScratchAddons/ScratchAddons] Adding back interesting 1.4 "image" functions (Discussion #3568)


I would love to be able to use in Scratch 3 some very useful Scratch 1.4 "image" functions (maybe useful just for Scratch teachers? I'm not sure, but, as a teacher, when I have taught Scratch to young kids and to teachers they were really useful) you had in your right-click menu when clicking the Stage or sprites on the Stage, namely:

  *   when right-clicking the Stage: 1) "grab screen region for new sprite" and 2) "save picture of Stage"
  *   when right-clicking a sprite on the Stage: 1) "grab screen region for new costume"

**Reply by sfederici** [reply]
> What is grabbing a screen region for new sprite?

Selecting a rectangular area as the costume of a new sprite.

**Reply by Samq64** [reply]
>All canvas elements by default have "Export image" right click options.

Yes, but it doesn't work. It just gives me a black rectangle even if there are obviously sprites on the stage.

**Reply by sfederici** [reply]
> In edge at least, exporting the stage is already possible, as all canvas elements by default have "Export image" right click options. 

Sorry, I had missed this part of your answer. Both in Edge and Chrome with ScratchAddons I can export the whole image of the canvas. What is convenient in Scratch 1.4 is that you can select just a region of the canvas via the selection rectangle and that what you have selected automatically become a new sprite or a new costume of the sprite selected in the sprite corral.
